ui/appui-react/src/appui-react/preview/control-widget-visibility/AddTabButton.tsx

```diff
@@ -0,0 +1,91 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Widget
+ */
+
+import * as React from "react";
+import { usePreviewFeatures } from "../PreviewFeatures";
+import { DropdownMenu, MenuItem } from "@itwin/itwinui-react";
+import { TabBarButton } from "../../layout/widget/Button";
+import { useLayout } from "../../layout/base/LayoutStore";
+import { getTabLocation } from "../../layout/state/TabLocation";
+import { SvgAdd } from "@itwin/itwinui-icons-react";
+import { NineZoneDispatchContext } from "../../layout/base/NineZone";
+import { WidgetIdContext } from "../../layout/widget/Widget";
+import { WidgetActionDropdownContext } from "../widget-action-dropdown/MoreButton";
+
+/** @internal */
+export function AddTabButton() {
+  const dispatch = React.useContext(NineZoneDispatchContext);
+  const widgetId = React.useContext(WidgetIdContext);
+  const dropdownContext = React.useContext(WidgetActionDropdownContext);
+  const tabs = useHiddenTabs();
+
+  if (!widgetId) return null;
+
+  const getMenuItems = (close?: () => void) =>
+    tabs.map((tab) => {
+      return (
+        <MenuItem
+          key={tab.id}
+          onClick={() => {
+            dispatch({
+              type: "WIDGET_TAB_ADD_TO_WIDGET",
+              id: tab.id,
+              widgetId,
+            });
+            dispatch({
+              type: "WIDGET_TAB_OPEN",
+              id: tab.id,
+            });
+            close?.();
+          }}
+        >
+          {tab.label}
+        </MenuItem>
+      );
+    });
+
+  const title = "Add tab";
```

Comment:
Should we change this to `Add widget` instead?

Reply:
Is `widget` considered the whole window including tabs?

Reply:
No, a widget maps to a single tab in the UI. Here's the latest documentation: https://github.com/iTwin/itwinjs-core/blob/ui-docs/docs/ui/appui/configure-frontstage.md#standard-frontstage-layout

Reply:
I see, in that case I would say `Add/Close widget` makes sense but I don't think `Add/Close tab` is bad because it also makes sense since these actions are perfomed in a widget container with other tabs already open. However, I think we should definitely have `Add widget` instead of `Add tab` in `NavigationWidget` (because it will show up if no widgets are open if I understand correctly?). Then maybe it would make sense to keep it consistent and have `Add/Close widget` everywhere?

Reply:
I agree, fixed as suggested.